api/v1_projects.inc

```diff
@@ -280,8 +285,8 @@ function render_project_json($project, $return_fields = null)
             "post_processor" => $project->PPer,
             "post_process_verifier" => $project->PPVer,
             "state" => $project->state,
-            "pages_available" => $project->n_available_pages,
-            "pages_total" => $project->n_pages,
+            "pages_available" => (int)$project->n_available_pages,
+            "pages_total" => (int)$project->n_pages,
```

Comment:
So these properties at runtime are string or does the encoder coerce everything to string if possible?

Reply:
It *appears* as though the mysqli driver returns all values to PHP as strings, which in most cases is fine since PHP "does the right thing" with numeric strings most of the time except that the `json_encode()` also treats them as strings unless we cast them otherwise:
```php
<?php
$relPath = "./";
include_once($relPath."base.inc");
include_once($relPath."Project.inc");

$projectid = 'projectID45c225f598e32';
$sql = sprintf("
    SELECT projectid, n_pages, postednum
    FROM projects
    WHERE projectid = '%s'",
    DPDatabase::escape($projectid)
);
$res = DPDatabase::query($sql);
$row = mysqli_fetch_assoc($res);
echo json_encode($row) . "\n";
```
returns
```json
{"projectid":"projectID45c225f598e32","n_pages":"418","postednum":null}
```
despite the `n_pages` column being defined as an int:
```
| n_pages                | smallint unsigned  | NO   |     | 0        |       |
```

Reply:
You're right! Check the recommendation for example 3 here: https://www.php.net/manual/en/mysqli.quickstart.statements.php
That is very surprising but I guess PHP typing is relatively newer. Glad you confirmed though.